configman/value_sources/for_modules.py

```diff
@@ -189,10 +196,40 @@ def local_to_str(a_thing):
 
 
 #==============================================================================
+class OrderableObj(object):
+    """Python3 can't sort non-string types implicitly.
+    """
+    def __init__(self, value):
+        if not isinstance(value, six.string_types):
+            value_str = to_str(value)
+        else:
+            value_str = value
+        self.value_str = value_str
+        self.value = value
+    def __lt__(self, other):
+        return (self.value_str < other.value_str)
+    def __repr__(self):
+        return self.value
+
+
+#==============================================================================
+class OrderableTuple(object):
+    """Python3 can't sort non-string types implicitly.
+    """
+    def __init__(self, value, index=1):
+        self.value = value
+        self.index = index
+    def __lt__(self, other):
+        return (self.value[self.index] < other.value[self.index])
+    def __repr__(self):
+        return self.value
```

Comment:
`functools.total_ordering` isn't in python 2.6.  We can skip it, add it and drop 2.6 support, or write our own total_ordering decorator that passes through in python>=2.7 and does something else in 2.6 (noop? roll our own `total_ordering`?)

Admittedly, I've never used `functools.total_ordering`, and these classes work without it.  OOC, what benefit will it provide?